### DISCUSSION TITLE
How do you suggest dealing with CORS protection in the browser while running this?

### DISCUSSION THREAD

**Discussion by Slider2k** [LostRuins/lite.koboldai.net#52] [Q&A] [answered]
<Subject>

**Comment by LostRuins** [comment]
While running... what? I need more details.

**Reply by Slider2k** [reply]
While running this as a standalone HTML.

**Reply by LostRuins** [reply]
Generally, the only way is to either disable CORS server side, or use a CORS proxy. What are you trying to connect to?

**Reply by Slider2k** [reply]
Just opening the the local html in the browser and connecting to whatever endpoint. Doesn't seem to be easily possible. One way I've found is disabling CORS security completely for the whole browser run with the cmd line argument, which is obviously not reasonable for normal operation. If only there was an easy way to disable CORS for select sites.

**Reply by LostRuins** [reply]
There is! It's usually a server configuration.

**Reply by Slider2k** [reply]
I'm sure there is a configuration for a **server**. But I'm talking about opening the html from a local drive on the desktop.

**Comment by henk717** [comment, marked as answer]
> I'm sure there is a configuration a **server**. But I'm talking about opening the html from a local drive on the desktop.

I think you misunderstand, the CORS behavior is decided by what you are connecting to not by us. So if you are using a specific API that does not allow connections from a local file the only thing you can do is connect to another provider that is less strict about ir. The AI Horde and local AI software such as Koboldcpp should not be strict about this. If you have a strong enough PC and value working offline I recommend giving Koboldcpp a try.

**Reply by Slider2k** [reply]
My mistake, I thought CORS was set on the html serving server.

**Reply by henk717** [reply]
In your case you definately want to switch to Koboldcpp, that way its already preconfigured and bundled and ready to go straight away. https://koboldai.org/cpp you will be able to use the same models (And technically even more since we preserved legacy support) as you can with llamacpp server, but the backend is custom made to work the best with all the features KoboldAI Lite has to offer and it will host its own bundled copy of KoboldAI Lite so you don't run in to this CORS issue.

**Reply by Slider2k** [reply]
Thanks, but I was more interested in a general solution of using a local single file html frontend with any supported backend. In that case using a reverse-proxy that inserts needed CORS policy is probably a way to go.
